File: internal/ui/src/services/backend.js
// Placeholder backend service using Wails calls via window.backend
export async function loadProjects() {
  try {
    return await window.backend.loadProjects();
  } catch (e) {
    console.error(e);
    return [];
  }
}

export async function saveProject(project) {
  try {
    return await window.backend.saveProject(project);
  } catch (e) {
    console.error(e);
    return project;
  }
}

export async function saveProjectData(projectId, data) {
  try {
    return await window.backend.saveProjectData(projectId, data);
  } catch (e) {
    console.error(e);
  }
}

File: internal/backend/backend.go
package backend

import "baristeuer/internal/data"

type Backend struct{}

func NewBackend() *Backend { return &Backend{} }

func (b *Backend) LoadProjects() []data.Project {
    return data.LoadProjects()
}

func (b *Backend) SaveProject(p data.Project) data.Project {
    return data.SaveProject(p)
}

func (b *Backend) SaveProjectData(id int, d data.ProjectData) {
    d.ProjectID = id
    data.SaveProjectData(d)
}
